### PR TITLE
Use simple-ember-auth for more robust authentication and session management

### PR DESCRIPTION
**Pull request by ConnorCimowsky**
Also, rip out js-cookie

bugs-bunny-florida

### CODE REVIEW

**Comment by ConnorCimowsky** [comment]
@Ommy, you should probably look at this since you did the login stuff :innocent:

File: src-frontend/app/authenticators/token.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                method: 'POST',
+                url: ENV.APP.API_HOST + '/auth/login',
+                data: credentials
+            }).then(function(response) {
+                Ember.run(function() {
+                    resolve({ token: response.token });
+                });
+            }, function(xhr, status, error) {
+                Ember.run(function() {
+                    reject(xhr);
```
Comment (Ommy) [inline src-frontend/app/authenticators/token.js]: What does the reject function do?

File: src-frontend/app/authenticators/token.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                reject();
+            }
+        });
+    },
+
+    authenticate(credentials) {
+        return new Ember.RSVP.Promise(function(resolve, reject) {
+            Ember.$.ajax({
+                method: 'POST',
+                url: ENV.APP.API_HOST + '/auth/login',
```
Comment (Trinovantes) [inline src-frontend/app/authenticators/token.js]: maybe the login url should be a constant in the ENV?

File: src-frontend/app/authorizers/token.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import Base from 'simple-auth/authorizers/base';
+
+export default Base.extend({
+    authorize: function(jqXHR, requestOptions) {
+        var session = this.get('session');
+
+        if (session.isAuthenticated) {
```
Comment (Trinovantes) [inline src-frontend/app/authorizers/token.js]: how does it know it's authenticated?

File: src-frontend/bower.json [hunk 4]
```diff
@@ -11,9 +11,9 @@
     "ember-resolver": "~0.1.15",
     "jquery": "^1.11.1",
     "jquery-ui": "~1.11.4",
-    "js-cookie": "^1.5.1",
     "loader.js": "ember-cli/loader.js#3.2.0",
     "qunit": "~1.17.1",
-    "bootstrap": "~3.3.4"
+    "bootstrap": "~3.3.4",
+    "ember-simple-auth": "0.8.0"
```
Comment (Trinovantes) [inline src-frontend/bower.json]: we should end this line with a comma so if we need to add more libraries, it won't show up as a deletion and insertion

**Comment by Ommy** [comment]
LGTM after questions

**Comment by Kapin** [comment]
LGTM once comments from others are addressed.

File: src-frontend/app/authenticators/token.js (hunk 1 above)
Comment (ConnorCimowsky) [inline src-frontend/app/authenticators/token.js]: This causes the promise to go unfulfilled, which tells `ember-simple-auth` that authentication was unsuccessful. We don't currently do anything in this case, but we already have a TODO for showing a banner or something.

File: src-frontend/app/authenticators/token.js (hunk 2 above)
Comment (ConnorCimowsky) [inline src-frontend/app/authenticators/token.js]: Agreed

File: src-frontend/app/authorizers/token.js (hunk 3 above)
Comment (ConnorCimowsky) [inline src-frontend/app/authorizers/token.js]: A successful authentication in `app/authenticators/token.js` will cause this to be set to `true` automatically :)

File: src-frontend/bower.json (hunk 4 above)
Comment (ConnorCimowsky) [inline src-frontend/bower.json]: :ok:

**Pull request merged**